client/src/components/timer.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react"
+import { useNavigate } from "react-router-dom"
+
+const msec2second = (msec: number) :number => {
+    return Math.floor(msec/1000)
+}
+
+// 現在時刻との差分を取ることで残り秒数(msec)を計算
+const calcRemain = (expire: Date) : number => {
+    const now = new Date()
+    const remain = expire.getTime() - now.getTime()
+
+    return remain
+}
+
+const Timer = ({expire} : {expire: Date}) => {
+    const [remain, setRemain] = useState(100)
+    // 時間切れ
+    const navigate = useNavigate()
+    const onTimeUp = () => {
+        navigate("/voting")
+    }
+
+    // mount時処理
+    useEffect(() => {
+        // 最初のレンダリング
+        setRemain(() => calcRemain(expire))
```

Comment:
これってnumberじゃなくてfunction渡してるけどエラー出ないんだ
```suggestion
        setRemain(calcRemain(expire))
```

Reply:
次のcommitで修正